### DISCUSSION TITLE
Support resize with a target file/stream size, rather than dimensions

### DISCUSSION THREAD

**Discussion by jimmcslim** [SixLabors/ImageSharp#1305] [Ideas] [answered]
A resize transform that targeted a maximum file size (possibly with dimension vs quality tradeoffs) would be appreciated!

**Comment by JimBobSquarePants** [comment, marked as answer]
I'm afraid that cannot be done; there's simply too many variables. 

Operations on images are performed on the uncompressed pixels laid out in memory. An encoded image is subject to the compression rules and constraints of the individual format. 

Say I resize two separate images to a given set of dimensions and encode them using the same image format and options. Unless those two images were identical you will result in two output images of different file size. 

You'd have to brute force, rinsing and repeating the resize + encoding operation over and over again. It's not efficient. 

Perhaps someone could build something nifty with machine learning that could make a best guess after training but that's well out of scope of the library.

**Reply by antonfirsov** [reply]
> Perhaps someone could build something nifty with machine learning that could make a best guess after training but that's well out of scope of the library.

Agree, this is how it should be done. Shouldn't be hard for someone experienced with machine learning, but unfortunately there are no data scientists / AI experts on board  @ SixLabors :)
